### PR TITLE
Added missing PATCH operation to Api Issue resource

### PR DESCRIPTION
**Pull request by promoisha**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.

### CODE REVIEW

**Review by solankisamir — CHANGES_REQUESTED** [review]
- Please rebase your branch, 2018-01-01 changes went into the stable branch folder.
- also you would need to update the 2018-06-01-preview, as that is files, we will use to create our next stable version. Also currently documentation is driven out of 2018-06-01-preview version.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/apimapis.json [hunk 1]
```diff
@@ -2857,6 +2857,74 @@
...
         }
       },
+      "patch": {
+        "tags": [
+          "ApiIssues"
+        ],
+        "operationId": "ApiIssue_Update",
+        "description": "Updates an existing issue for an API.",
+        "x-ms-examples": {
+          "ApiManagementCreateApiIssue": {
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/apimapis.json]: fix Example Name `ApiManagementUpdateApiIssue`

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-06-01-preview/apimapis.json [hunk 2]
```diff
@@ -2645,6 +2645,65 @@
...
+          },
+          {
+            "$ref": "#/parameters/IssueIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/IssueContract"
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-06-01-preview/apimapis.json]: For Update you will need to create a separate contract, as the IssueContract has the following required properties
`title`
`description`
`userId`

Maybe you can take out commonn properties into base properties.

**Comment by annatisch** [comment]
Thanks @promoisha - please rebase your branch, as there are merge conflicts. :)

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2018-01-01/apimapis.json [hunk 3]
```diff
@@ -4549,6 +4608,16 @@
       ],
       "description": "Issue Contract details."
     },
+    "IssueUpdateContract": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/IssueContractUpdateProperties",
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2018-01-01/apimapis.json]: nit: IssueUpdateContractProperties

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2018-01-01/examples/ApiManagementUpdateApiIssue.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+    "parameters": {
+      "serviceName": "apimService1",
+      "resourceGroupName": "rg1",
+      "api-version": "2017-03-01",
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2018-01-01/examples/ApiManagementUpdateApiIssue.json]: fix version in example. 2018-01-01

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-06-01-preview/examples/ApiManagementUpdateApiIssue.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+    "parameters": {
+      "serviceName": "apimService1",
+      "resourceGroupName": "rg1",
+      "api-version": "2017-03-01",
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-06-01-preview/examples/ApiManagementUpdateApiIssue.json]: fix version 2018-06-01-preview

**Review by solankisamir — CHANGES_REQUESTED** [review]

**Review by solankisamir — APPROVED** [review]
:shipit:

**Review by annatisch — APPROVED** [review]

**Pull request merged**